pyro/generic/testing.py

```diff
@@ -0,0 +1,106 @@
+import argparse
+from collections import OrderedDict
+
+from pyro.generic import distributions as dist, handlers
+from pyro.generic import infer, ops, pyro, pyro_backend
+
+
+MODELS = OrderedDict()
+
+
+def register(name):
+    def _append(fn):
+        MODELS[name] = fn
+    return _append
+
+
+@register('logistic_regression')
+def logistic_regression():
+    N, dim = 3000, 3
+    # generic way to sample from distributions
+    data = pyro.sample('data', dist.Normal(0., 1.), sample_shape=(N, dim))
+    true_coefs = ops.arange(1., dim + 1.)
+    logits = ops.sum(true_coefs * data, axis=-1)
+    labels = pyro.sample('labels', dist.Bernoulli(logits=logits))
+
+    def model(x, y):
+        coefs = pyro.sample('coefs', dist.Normal(ops.zeros(dim), ops.ones(dim)))
+        intercept = pyro.sample('intercept', dist.Normal(0., 1.))
+        logits = ops.sum(coefs * x, axis=-1) + intercept
+        return pyro.sample('obs', dist.Bernoulli(logits=logits), obs=y)
+
+    return model, (data, labels), {}
```

Comment:
Could we make these dicts so as to be forwards compatible with SVI and special forms of testing? E.g.
```py
return {'model': model, 'model_args': (data, labels)}
```

Reply:
That way we can add things like `'guide': guide` in the future.

Reply:
Sounds good!